index.js

```diff
@@ -57,45 +57,19 @@ function valueToChangeAddItem(value, index) {
   };
 }
 
-/*
-  There are two common cases when we should treat additions as adjacent:
-
-    1. two adjacent additions (i.e. contiguous indexes);
-    2. two additions separated by a deletion.
-
-  The second case shall occur, perhaps often, when using the splice method or
-  redefining an array.
-
-  One could theoretically walk backwards through the changes and form a longest-
-  contiguous list of additions, but that *could* be O(n)ish worst case (though
-  average case is probably quite good).
-  */
-function isAdditionAdjacentToLast(changeIndex, arrayChanges) {
-  var change = arrayChanges[changeIndex];
-  var prevChange = arrayChanges[changeIndex - 1];
-
-  if (changeIndex === 0 ||
-      changeIndex >= arrayChanges.length ||
-      change.status !== 'added'
-    ) { return false; }
-
-  // Add-Add
-  if (prevChange.status === 'added' && prevChange.index === change.index - 1) {
-    return true;
+function arrayClear(array) {
+  while (array.length) {
```

Comment:
Ah.. yes I know about it, just didn't come to my mind at that moment. I will refactor it.